### PR TITLE
Add Lambda init debug mode

### PR DESCRIPTION
**Pull request by dominikschubert**

### CODE REVIEW

File: localstack/config.py [hunk 1]
```diff
@@ -813,6 +822,10 @@ def in_docker():
     "LAMBDA_EXECUTOR",
     "LAMBDA_FALLBACK_URL",
     "LAMBDA_FORWARD_URL",
+    "LAMBDA_INIT_DEBUG",
+    "LAMBDA_INIT_BIN_PATH",
+    "LAMBDA_INIT_DELVE_PATH",
+    "LAMBDA_INIT_DELVE_PORT",
```
Comment (dominikschubert) [inline localstack/config.py:828]: one is still missing here

File: localstack/config.py [hunk 2]
```diff
@@ -680,6 +680,15 @@ def in_docker():
 # reach the container via its IPv4 (e.g., macOS https://docs.docker.com/desktop/networking/#i-cannot-ping-my-containers)
 LAMBDA_DEV_PORT_EXPOSE = is_env_true("LAMBDA_DEV_PORT_EXPOSE")
 
+# INTERNAL debugging options
+# There are NO stability guarantees and they may break at any time.
+# LAMBDA_INIT_VERBOSE = is_env_true("LAMBDA_INIT_VERBOSE")
+LAMBDA_INIT_DEBUG = is_env_true("LAMBDA_INIT_DEBUG")
```
[the remark this answers is not in the document]
Comment (joe4dev) [reply localstack/config.py:686]: not so important though; rather a convenience option

**Review by joe4dev — APPROVED** [review]
Very helpful debug mode 👍 
Just a bit verbose configuration.

File: localstack/config.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (dominikschubert) [reply localstack/config.py:686]: Hm I was thinking about this and for now tried to go with the more explicit/verbose option to not make too many  assumptions on how we're going to use this. In a month or so we can reconsider though :grin:

File: localstack/services/awslambda/invocation/docker_runtime_executor.py [hunk 3]
```diff
@@ -307,6 +314,18 @@ def start(self, env_vars: dict[str, str]) -> None:
             CONTAINER_CLIENT.copy_into_container(
                 self.container_name, f"{str(get_runtime_client_path())}/.", "/"
             )
+            # tiny bit inefficient since we actually overwrite the init, but otherwise the path might not exist
+            if config.LAMBDA_INIT_DEBUG:
```
[the remark this answers is not in the document]
Comment (dominikschubert) [reply localstack/services/awslambda/invocation/docker_runtime_executor.py:318]: Yeah there are quite a few options:

* Change the entrypoint script (what you linked)
* Add a way to skip the modification of the entrypoint in the docker runtime executor. 
* completely separate code path that only copies the custom lambda init

The first option won't need any changes here and is transparent to localstack, so I'd prefer that for now :thinking: It'll also allow reconfiguring this "on-the-fly" while the instance is running.

**Review by dfangl — APPROVED** [review]
LGTM!

**Pull request merged**